### ISSUE TITLE
Parameter Count seems failed in destructuring object parameters

### ISSUE THREAD

**Issue by zymbth** [rrd108/vue-mess-detector#365]
**Describe the bug**
[Parameter Count](https://vue-mess-detector.webmania.cc/rules/rrd/parameter-count.html) seems failed in destructuring object parameters

**To Reproduce**

1. What is the exact code you are analyzing?

```js
function dummyFuncOne({ param1, param2, param3, param4, param5 }) {
  return 'One'
}

const dummyFuncTwo = ({ param1, param2, param3, param4 }) => {
  return 'Two'
}
```

2. What is the output you are getting?

```text
-  rrd ~ parameter count
   src\views\Test\parameter-count.js
   👉 Max number of function parameters should be 3. See: https://vue-mess-detector.webmania.cc/rules/rrd/parameter-count.html
   function dummyFuncOne has 5 parameters 🚨

   src\views\Test\parameter-count.js
   👉 Max number of function parameters should be 3. See: https://vue-mess-detector.webmania.cc/rules/rrd/parameter-count.html
   function dummyFuncTwo has 4 parameters 🚨
```

**Expected behavior**
Correctly count the destructured object parameters.

**Screenshots**
![image](https://github.com/user-attachments/assets/c5dcd564-f603-441e-89ef-5ae2e7f31956)

**Used version number of vue-mess-detector:** `0.57.0`

**Used version number of node & yarn:** `node@18.20.4`, `pnpm@9.8.0`

**Comment by rrd108** [comment]
@David-Pena it is like that by design, isn't it?

From the code maintainability viewpoint they have 4 and 5 parameters.

I understand the issue, but I want to be sure how we think about this and clearly state that in the docs.

**Comment by David-Pena** [comment]
> it is like that by design, isn't it?

I don't think so, right now I don't remember completely but I think its just misssing the correct logic to identify destructured params

> From the code maintainability viewpoint they have 4 and 5 parameters.

I think destructured props shouldnt be limited. In this case its only one param so it shouldnt be reported, what do you think? @rrd108

**Comment by rrd108** [comment]
When the param is an object - I think still there is a reasonable limit of *non optional* properties

but for now I am ok handling it as one

**Comment by David-Pena** [comment]
> When the param is an object - I think still there is a reasonable limit of non optional properties

Lets create a separate issue for that change to the rule

**Comment by David-Pena** [comment]
Do you have any progress in here? @rrd108 

@zymbth would you like to send a PR?

**Comment by zymbth** [comment]
@David-Pena Sorry for the late reply, just got back from vacation. I've been a bit busy lately, try to contribute in the future.

**Comment by David-Pena** [comment]
This is solved in https://github.com/rrd108/vue-mess-detector/commit/bbded51c4de87bb8e0f24c90e334b01b570e5b43
